File: lab1/src/main/java/pl/pjatk/bmi/MainActivity.java
package pl.pjatk.bmi;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Locale;

public class MainActivity extends AppCompatActivity {

    private TextView height;
    private TextView weight;
    private LinearLayout resultComponent;
    private TextView result;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        height = findViewById(R.id.inputHeight);
        weight = findViewById(R.id.inputWeight);
        resultComponent = findViewById(R.id.resultComponent);
        result = findViewById(R.id.result);

        Button calculate = findViewById(R.id.calculate);
        Button reset = findViewById(R.id.reset);

        calculate.setOnClickListener(v -> {
            if (isValidated(height, weight)) {
                calculate();
            }
        });

        reset.setOnClickListener(v -> {
            height.setText(null);
            height.setError(null);
            weight.setText(null);
            weight.setError(null);

            result.setText(null);
            resultComponent.setVisibility(View.GONE);
        });
    }

    /**
     * validates views and setting errors in case of wrong input
     * @param views input views
     * @return true/false
     */
    private boolean isValidated(final View...views) {
        boolean validated = true;
        for (View view : views) {
            TextView v = (TextView) view;
            if (v.getText().toString().length() == 0) {
                v.setError("Field is required");
                validated = false;
            } else if (!(Float.parseFloat(v.getText().toString()) > 0)) {
                v.setError("Value should be a positive number");
                validated = false;
            }
        }
        return validated;
    }

    /**
     * calculates BMI with formula weight/height^2
     * input units:
     *  height - centimeters [converted to meters]
     *  weight - kilograms
     */
    private void calculate() {

        float height = -1;
        float weight = -1;
        try {
            height = Float.parseFloat(this.height.getText().toString());
            weight = Float.parseFloat(this.weight.getText().toString());
        } catch (NumberFormatException e) {
            System.err.println("Wrong input");
        }

        Double res = weight / Math.pow(height/100, 2);

        result.setText(String.format(Locale.getDefault(), "%.2f", res));
        resultComponent.setVisibility(View.VISIBLE);
    }
}

File: lab3/src/main/java/pl/pjatk/lab3/activity/MainActivity.kt
package pl.pjatk.lab3.activity

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import pl.pjatk.lab3.R

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val bmiButton: Button = findViewById(R.id.btn_activity_bmi)
        bmiButton.setOnClickListener { runActivity(BmiActivity::class.java) }
        val chartButton: Button = findViewById(R.id.btn_activity_chart)
        chartButton.setOnClickListener { runActivity(ChartActivity::class.java) }
        val quizButton: Button = findViewById(R.id.btn_activity_quiz)
        quizButton.setOnClickListener { runActivity(QuizActivity::class.java) }
    }

    private fun runActivity(clazz: Class<*>) {
        val intent = Intent(this, clazz)
        startActivity(intent)
    }
}
